src/components/moviesSection/moviesSection.tsx

```diff
@@ -10,19 +13,35 @@ import styles from './moviesSection.module.css'
 
 export const MoviesSection: React.FC = () => {
   const [moviesData, setMoviesData] = React.useState(movies)
+  const [selectedMovieName, setSelectedMovieName] = React.useState('')
+  const [selectedMovieYear, setSelectedMovieYear] = React.useState(0)
+
+  const { isDisplayed, toggleModal } = useModal()
 
   const handleSort = () => {
     const moviesDataCopy = [...moviesData]
     moviesDataCopy.sort(sortByYear)
     setMoviesData(moviesDataCopy)
   }
 
+  const handleMovieClick = (e: any) => {
+    const selectedMovieId = e.currentTarget.dataset.id
+    const selectedMovie = moviesData.find(movie => movie.id == selectedMovieId)
+
+    setSelectedMovieName(selectedMovie.name)
+    setSelectedMovieYear(selectedMovie.year)
+    toggleModal()
+  }
```

Comment:
Change to something like this with typing and fix when you compare number id to string id. Also, use strict comparison

```suggestion
  const handleMovieClick = (e: React.SyntheticEvent<HTMLElement>) => {
    const selectedMovieId = e.currentTarget.dataset.id
    const { name, year } = moviesData.find(
      ({ id }) => String(id) === selectedMovieId,
    )

    setSelectedMovieName(name)
    setSelectedMovieYear(year)
    toggleModal()
  }
```